web/source/osk/visualKeyboard.ts

```diff
@@ -1710,10 +1710,12 @@ namespace com.keyman.osk {
 
       if(usePreview) {
         // Previews are not permitted for keys using any of the following CSS styles.
-        var excludedClasses = ['kmw-key-shift',    // special keys
-                               'kmw-key-shift-on', // active special keys (shift, when in shift layer
-                               'kmw-spacebar',     // space
-                               'kmw-key-blank',    // Keys that are only used for layout control
+        var excludedClasses = ['kmw-key-shift',      // special keys
+                               'kmw-key-shift-on',   // active special keys (shift, when in shift layer
+                               'kmw-key-special',    // special keys that require the keyboard's OSK font
+                               'kmw-key-special-on', // active special keys requiring the keyboard's OSK font
+                               'kmw-spacebar',       // space
+                               'kmw-key-blank',      // Keys that are only used for layout control
```

Comment:
For comparison:  https://github.com/keymanapp/keyman/blob/cb98febc206c7e2d83030bb8b4890d58534cc7e1/common/core/web/keyboard-processor/src/keyboards/activeLayout.ts#L157-L171

So, the end-user difference in 'special' vs 'shift' is meant to come down to font control.  Unfortunately, this detail was ignored by the actual key preview logic.